backend/substrapp/tasks/k8s_backend.py

```diff
@@ -144,6 +152,13 @@ def watch_pod(name, watch_init_container=False):
             attempt += 1
             logger.error(f'Could not get pod "{name}" status (attempt {attempt}/{max_attempts}): {e}')
 
+    # Merge error
+    if error_pod is not None:
```

Comment:
What's the reason for not setting the `error` variable inside the `if api_response.status.phase == 'Failed' or (api_response.status.reason and 'Evicted' in api_response.status.reason):` condition ?

To do that, maybe we could move the if condition to the end of code inside the while loop?

Reply:
I change the code to be simpler, so it's not necessary to have this merge code :)

Reply:
Thanks!